### PR TITLE
Migration flow: Fix migrate ready screen event triggering issue

### PR DESCRIPTION
**Pull request by ouikhuan**
Related to #82923 

## Proposed Changes

* Currently the `calypso_site_migration_ready_screen` event gets triggered when the Jetpack install screen shows up and this is not the expected behavior. The root cause is because `requiresPluginUpdate` might be `false` when first rendered and change to `true` later. We change and use `siteCanMigrate === true` instead so we can make sure we get the correct status. We also move `migrationTrackingProps` to useMemo so it won't be identified as a new object on every render and triggers the event multiple times.

## Testing Instructions

*  Navigate to http://calypso.localhost:3000/setup/import-focused/import?siteSlug=${target_site}
*  Fill in a source site without the Jetpack connection and go through the flow until you're on the Jetpack install screen, check and make sure only `calypso_site_importer_show_update_info` event gets triggered.
* Try to connect the Jetpack during the flow and once the screen switches to the migrate ready screen `calypso_site_migration_ready_screen` should get triggered.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/blocks/importer/wordpress/import-everything/pre-migration/index.tsx
Comment (bogiii) [inline client/blocks/importer/wordpress/import-everything/pre-migration/index.tsx]: After merging https://github.com/Automattic/wp-calypso/pull/83088, we can remove this entire `useEffect` block and move triggering the "calypso_site_migration_ready_screen" event to the MigrationReady component.

**Review by bogiii — APPROVED** [review]
Works as described!

File: client/blocks/importer/wordpress/import-everything/pre-migration/index.tsx
Comment (ouikhuan) [reply client/blocks/importer/wordpress/import-everything/pre-migration/index.tsx]: That's what I'm thinking as well, I'll hold this PR for now until #83088 is merged!

Comment (ouikhuan) [reply client/blocks/importer/wordpress/import-everything/pre-migration/index.tsx]: @bogiii I've move the useEffect into it's own component, could you review it again?

**Review by bogiii — APPROVED** [review]
Perfect! 🚢

**Pull request merged**
